src/3rd_party/CMakeLists.txt

```diff
@@ -283,23 +283,24 @@ if(HMI_DBUS_API)
   add_subdirectory(dbus-cmake)
 endif()
 
-set(FORCE_BSON_REBUILD FALSE CACHE INTERNAL "Force rebuild of bson library" FORCE)
-set(BSON_LIBS_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/lib CACHE INTERNAL "Instalation path of bson libraries" FORCE)
-set(BSON_INCLUDE_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/include CACHE INTERNAL "Instalation path of bson headers" FORCE)
-set(EMHASHMAP_LIBS_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/lib CACHE INTERNAL "Instalation path of emashmap libraries" FORCE)
-set(EMHASHMAP_INCLUDE_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/include CACHE INTERNAL "Instalation path of emashmap headers" FORCE)
-set(BSON_LIB_SOURCE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bson_c_lib CACHE INTERNAL "Sources of bson library" FORCE)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}")
+set(CMAKE_SOURCE_PREFIX ${CMAKE_SOURCE_PREFIX} "${3RD_PARTY_INSTALL_PREFIX}")
+find_package (BSON)
+message (STATUS "bson installed in " ${BSON_LIBS_DIRECTORY} " , " ${BSON_INCLUDE_DIRECTORY})
+message (STATUS "emhashmap installed in " ${EMHASHMAP_LIBS_DIRECTORY} " , " ${EMHASHMAP_INCLUDE_DIRECTORY})
 
-IF (NOT EXISTS ${BSON_LIBS_DIRECTORY}/libbson.so OR (${FORCE_BSON_REBUILD}))
-  IF (${FORCE_BSON_REBUILD})
-    MESSAGE(INFO "Force rebuild bson lib")
-  endif()
-
-  add_custom_target(clean_bson_build_dir
-    COMMAND git clean -dfx && echo "Clean bson dir"
-    WORKING_DIRECTORY ${BSON_LIB_SOURCE_DIRECTORY}
-    )
+if (${BSON_LIB} MATCHES "BSON_LIB-NOTFOUND")
+  message (STATUS "Building bson required")
+  set(BSON_LIB_SOURCE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bson_c_lib CACHE INTERNAL "Sources of bson library" FORCE)
+  set(BSON_LIBS_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/lib CACHE INTERNAL "Instalation path of bson libraries" FORCE)
+  set(BSON_INCLUDE_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/include CACHE INTERNAL "Instalation path of bson headers" FORCE)
+  set(EMHASHMAP_LIBS_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/lib CACHE INTERNAL "Instalation path of emashmap libraries" FORCE)
+  set(EMHASHMAP_INCLUDE_DIRECTORY ${3RD_PARTY_INSTALL_PREFIX}/include CACHE INTERNAL "Instalation path of emashmap headers" FORCE)
 
+  set(BSON_INSTALL_COMMAND make install)
+  if (${3RD_PARTY_INSTALL_PREFIX} MATCHES "/usr/local")
+    set(BSON_INSTALL_COMMAND sudo make install)
+  endif()
   include(ExternalProject)
   ExternalProject_Add(libbson
       GIT_REPOSITORY "http://github.com/LuxoftAKutsan/bson_c_lib.git"
```

Comment:
@LuxoftAKutsan 
Leave a comment here why your personal repo is specified here and when its planned to removed it